cirq-google/cirq_google/calibration/phased_fsim_test.py

```diff
@@ -874,3 +875,40 @@ def test_options_phase_corrected_override():
         ).zeta_chi_gamma_correction_override()
         == PhasedFSimCharacterization()
     )
+
+
+def test_try_convert_gate_to_fsim():
+    fsim = cirq.FSimGate(theta=0.3, phi=0.5)
+    assert try_convert_gate_to_fsim(fsim) == PhaseCalibratedFSimGate(fsim, 0.0)
+
+    gate = cirq.ISwapPowGate(exponent=-0.5)
+    expected = PhaseCalibratedFSimGate(cirq.FSimGate(theta=0.25 * np.pi, phi=0.0), 0.0)
+    assert try_convert_gate_to_fsim(gate) == expected
+
+    gate = cirq.PhasedFSimGate(theta=0.2, phi=0.5, chi=1.5 * np.pi)
+    expected = PhaseCalibratedFSimGate(cirq.FSimGate(theta=0.2, phi=0.5), 0.25)
+    assert try_convert_gate_to_fsim(gate) == expected
+
+    gate = cirq.PhasedFSimGate(theta=0.2, phi=0.5, zeta=1.5 * np.pi)
+    assert try_convert_gate_to_fsim(gate) is None
+
+    gate = cirq.PhasedISwapPowGate(exponent=-0.5, phase_exponent=0.7)
+    expected = PhaseCalibratedFSimGate(cirq.FSimGate(theta=0.25 * np.pi, phi=0.0), -0.7)
+    assert try_convert_gate_to_fsim(gate) == expected
+
+    gate = cirq.CZ
+    expected = PhaseCalibratedFSimGate(cirq.FSimGate(theta=0.0, phi=np.pi), 0.0)
+    assert try_convert_gate_to_fsim(gate) == expected
+
+    gate = cirq.ops.CZPowGate(exponent=0.3)
+    expected = PhaseCalibratedFSimGate(cirq.FSimGate(theta=0.0, phi=-0.3 * np.pi), 0.0)
+    assert try_convert_gate_to_fsim(gate) == expected
+
+    gate = cirq.ops.CZPowGate(exponent=0.3, global_shift=0.4)
+    assert try_convert_gate_to_fsim(gate) is None
+
+    gate = cirq_google.ops.SYC
+    expected = PhaseCalibratedFSimGate(cirq_google.ops.SYC, 0.0)
+    assert try_convert_gate_to_fsim(gate) == expected
```

Comment:
How does this work? I don't see the SYC gate create for the new converter?

Reply:
This is because SycamoreGate inherits from FSimGate, so it is handled by the branch "if isinstance(gate, cirq.FSimGate)".
I added an explicit branch so it would produce cirq.FSimGate, not SycamoreGate - but I don't think it's necessary.